pkg/controller/sparkapplication/controller.go

```diff
@@ -546,63 +542,52 @@ func (c *Controller) submitSparkApplication(app *v1alpha1.SparkApplication) *v1a
 				State:        v1alpha1.FailedSubmissionState,
 				ErrorMessage: err.Error(),
 			},
-			SubmissionAttempts:        app.Status.SubmissionAttempts + 1,
-			LastSubmissionAttemptTime: metav1.Now(),
 		}
 		return app
 	}
 
-	// Try submitting the application by running spark-submit.
-	submitted, err := runSparkSubmit(newSubmission(submissionCmdArgs, appToSubmit))
+	_, err = c.subJobManager.createSubmissionJob(newSubmission(submissionCmdArgs, appToSubmit))
 	if err != nil {
-		app.Status = v1alpha1.SparkApplicationStatus{
-			AppState: v1alpha1.ApplicationState{
-				State:        v1alpha1.FailedSubmissionState,
-				ErrorMessage: err.Error(),
-			},
-			SubmissionAttempts:        app.Status.SubmissionAttempts + 1,
-			LastSubmissionAttemptTime: metav1.Now(),
+		if !errors.IsAlreadyExists(err) {
+			app.Status = v1alpha1.SparkApplicationStatus{
+				AppState: v1alpha1.ApplicationState{
+					State:        v1alpha1.FailedSubmissionState,
+					ErrorMessage: err.Error(),
+				},
+			}
 		}
-		c.recordSparkApplicationEvent(app)
-		glog.Errorf("failed to run spark-submit for SparkApplication %s/%s: %v", app.Namespace, app.Name, err)
 		return app
 	}
-	if !submitted {
-		// The application may not have been submitted even if err == nil, e.g., when some
-		// state update caused an attempt to re-submit the application, in which case no
-		// error gets returned from runSparkSubmit. If this is the case, we simply return.
-		return app
-	}
-
-	glog.Infof("SparkApplication %s/%s has been submitted", app.Namespace, app.Name)
+	glog.Infof("Submission Job for SparkApplication %s/%s has been created", app.Namespace, app.Name)
 	app.Status = v1alpha1.SparkApplicationStatus{
-		AppState: v1alpha1.ApplicationState{
-			State: v1alpha1.SubmittedState,
-		},
-		SubmissionAttempts:        app.Status.SubmissionAttempts + 1,
-		ExecutionAttempts:         app.Status.ExecutionAttempts + 1,
+		AppState:                  v1alpha1.ApplicationState{State: v1alpha1.PendingSubmissionState},
 		LastSubmissionAttemptTime: metav1.Now(),
```

Comment:
Because now Job controller drives submission-attempts, LastSubmissionAttemptTime  is probably not what we expect it to be ? Let's remove or rename this ?

Reply:
We can remove it.